tests/epics/adcore/test_single_trigger.py

```diff
@@ -1,42 +1,48 @@
+import bluesky.plan_stubs as bps
 import bluesky.plans as bp
 import pytest
 from bluesky import RunEngine
 
-from ophyd_async.core import DeviceCollector, set_mock_value
+import ophyd_async.plan_stubs as ops
 from ophyd_async.epics import adcore
 
 
 @pytest.fixture
-async def single_trigger_det():
-    async with DeviceCollector(mock=True):
-        stats = adcore.NDPluginStatsIO("PREFIX:STATS")
-        det = adcore.SingleTriggerDetector(
-            drv=adcore.ADBaseIO("PREFIX:DRV"),
-            stats=stats,
-            read_uncached=[stats.unique_id],
-        )
+async def single_trigger_det_with_stats():
+    stats = adcore.NDPluginStatsIO("PREFIX:STATS", name="stats")
+    det = adcore.SingleTriggerDetector(
+        drv=adcore.ADBaseIO("PREFIX:DRV"),
+        stats=stats,
+        read_uncached=[stats.unique_id],
+        name="det",
+    )
 
-    assert det.name == "det"
-    assert stats.name == "det-stats"
     # Set non-default values to check they are set back
     # These are using set_mock_value to simulate the backend IOC being setup
     # in a particular way, rather than values being set by the Ophyd signals
-    set_mock_value(det.drv.acquire_time, 0.5)
-    set_mock_value(det.drv.array_counter, 1)
-    set_mock_value(det.drv.image_mode, adcore.ImageMode.continuous)
-    set_mock_value(stats.unique_id, 3)
-    yield det
```

Comment:
Why did you have to move these into the plan?

Reply:
Ah right these shouldn't require a `connect` call first. I'll move them.

Reply:
Ah I remember why moved them, `set_mock_value` requires a `MockSignalBackend` and until these are connected (which has to be done in the plan, the backend isn't switched out.

Reply:
Ah, because these are connected in the RE, while they used to be connected under `DeviceCollector`. That makes sense.